src/plugins/dashboard/public/application/dashboard_app_controller.tsx

```diff
@@ -430,9 +430,16 @@ export class DashboardAppController {
               .getStateTransfer(scopedHistory())
               .getIncomingEmbeddablePackage();
             if (incomingState) {
-              container.addNewEmbeddable<SavedObjectEmbeddableInput>(incomingState.type, {
-                savedObjectId: incomingState.id,
-              });
+              if ('id' in incomingState) {
+                container.addNewEmbeddable<SavedObjectEmbeddableInput>(incomingState.type, {
+                  savedObjectId: incomingState.id,
+                });
+              } else if ('input' in incomingState) {
+                container.addNewEmbeddable<SavedObjectEmbeddableInput>(
```

Comment:
Should this type be `SavedObjectEmbeddableInput` instead of the more generic `EmbeddableInput`? It's strange to me that this type parameter doesn't seem to affect anything.